contribs/drt-extensions/src/main/java/org/matsim/contrib/drt/extension/insertion/objectives/VehicleDistanceObjective.java

```diff
@@ -0,0 +1,32 @@
+package org.matsim.contrib.drt.extension.insertion.objectives;
+
+import org.matsim.contrib.drt.extension.insertion.DrtInsertionObjective;
+import org.matsim.contrib.drt.extension.insertion.distances.DistanceCalculator;
+import org.matsim.contrib.drt.extension.insertion.distances.InsertionDistanceCalculator;
+import org.matsim.contrib.drt.extension.insertion.distances.InsertionDistanceCalculator.VehicleDistance;
+import org.matsim.contrib.drt.optimizer.insertion.InsertionDetourTimeCalculator.DetourTimeInfo;
+import org.matsim.contrib.drt.optimizer.insertion.InsertionGenerator.Insertion;
+import org.matsim.contrib.drt.passenger.DrtRequest;
+
+public class VehicleDistanceObjective implements DrtInsertionObjective {
```

Comment:
If this is the objective then probably the path search should also use the distance at the objective to keep things consistent.

Reply:
Hm, it depends on how you mean it. As long as the insertion algorithm is based on travel time, I think here we should also use travel time in the router when we "reconstruct" the routes that have been used to build the routes, to be coherent. Otherwise, I agree, if we can base the insertion on distances (see my other comment) then it would make sense to do that here as well. But then ideally we just have this information available in the detourInfo so we don't need to route at all at this point :)